### PR TITLE
Add animations

### PR DESCRIPTION
**Pull request by Boris-Em**
Add 4 types of animations.

### CODE REVIEW

File: ARCharts/ARBarChart.swift [hunk 1]
```diff
@@ -125,21 +129,36 @@ public class ARBarChart: SCNNode {
...
                 previousXPosition = xPosition
+                
+                if animationType == .grow || animationType == .progressiveGrow {
+                    let delay = animationType == .progressiveGrow ? Double(index) * animationDuration / 10.0 : nil
+                    let heightAnim = heightAnimation(withToValue: barHeight, duration: animationDuration, delay: delay)
+                    let yPositionAnim = yPositionAnimation(withToValue: yPosition, duration: animationDuration, delay: delay)
+                    barBox.addAnimation(heightAnim, forKey: "height")
+                    barNode.addAnimation(yPositionAnim, forKey: "position.y")
+                } else if animationType == .fadeIn || animationType == .progressiveFadeIn {
+                    let delay = animationType == .progressiveFadeIn ? Double(index) * animationDuration / 10.0 : nil
```
Comment (chrischute) [inline ARCharts/ARBarChart.swift]: Nice, this will be a cool effect.

File: SampleApp/ViewController.swift [hunk 2]
```diff
@@ -92,11 +92,14 @@ class ViewController: UIViewController, ARSCNViewDelegate {
...
-        let values = generateRandomNumbers(withRange: 0..<50, numberOfRows: 50, numberOfColumns: 50)
-            
+//        let values = generateRandomNumbers(withRange: 0..<50, numberOfRows: 50, numberOfColumns: 50)
+        let values = generateNumbers(fromDataSampleWithIndex: 1)!
+        
         let dataSeries = ARDataSeries(withValues: values)
         self.barChart = ARBarChart(dataSource: dataSeries, delegate: dataSeries, size: SCNVector3(0.2, 0.2, 0.2))
         self.barChart.position = position
+        self.barChart.animationType = .grow
+        self.barChart.drawGraph()
```
Comment (chrischute) [inline SampleApp/ViewController.swift]: Is there a way to keep `drawGraph` private, and just listen for when the bar chart gets added to a parent node?

**Review by chrischute — APPROVED** [review]
Mostly looks great. My only big suggestion is that we wrap up all the animation logic in the `AnimationManager` class. That is, make the `heightAnimation` and similar methods private, and give each bar chart an `AnimationManager` member, and then you'd call a method like `barChart.animationManager.animate(barNode, atIndex: index, forSeries: series)`. The manager would then attach the animations to `barNode` and `barBox` (which is just `barNode.geometry`). We could do this in a new PR if you agree.

File: SampleApp/ViewController.swift (hunk 2 above)
Comment (Boris-Em) [reply SampleApp/ViewController.swift]: That was my struggle for the longest time yesterday =/
It doesn't seem like it. I couldn't find any lifecycle functions to overwrite for `SCNNode`.

File: ARCharts/ARBarChart.swift [hunk 3]
```diff
@@ -125,21 +129,31 @@ public class ARBarChart: SCNNode {
...
+                barNode.position = SCNVector3(x: xPosition, y: startingYPosition, z: zPosition)
                 
                 let barColor = delegate.barChart(self, colorForBarAtIndex: index, forSeries: series)
                 barNode.geometry?.firstMaterial?.diffuse.contents = barColor
                 
                 self.addChildNode(barNode)
                 previousXPosition = xPosition
+                
+                if let animationType = animationType {
+                    let animationManager = AnimationManager(animationType: animationType, animationDuration: animationDuration)
```
Comment (chrischute) [inline ARCharts/ARBarChart.swift]: Can we construct the `AnimationManager` only once on the bar chart, when the `animationType` gets set?

File: ARCharts/ARBarChart.swift [hunk 4]
```diff
@@ -155,6 +161,11 @@ public class ARBarChart: SCNNode {
                     self.addLabel(forIndex: index, atXPosition: xPosition + xShift - 0.5 * barsWidth)
                 }
                 previousXPosition = xPosition
+                
+                if let animationType = animationType {
+                    let animationManager = AnimationManager(animationType: animationType, animationDuration: animationDuration)
+                    animationManager.addAnimation(toBarNode: barNode, atIndex: index, withBarHeight: barHeight, yPosition, opacity)
+                }
```
Comment (chrischute) [inline ARCharts/ARBarChart.swift]: Can we construct the `AnimationManager` only once, when the `animationType` gets set? I.e., make `animationManager` a private member on `ARBarChart`, which gets set on `didSet` on the `animationType`?

File: ARCharts/ARAnimationManager.swift [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+public enum ARBarChartAnimationType {
+    case fadeIn
+    case progressiveFadeIn
+    case grow
+    case progressiveGrow
+}
+
+struct AnimationManager {
+    
+    var animationType: ARBarChartAnimationType
```
Comment (chrischute) [inline ARCharts/ARAnimationManager.swift:21]: Nit: These could be `private let`s.

**Review by chrischute — APPROVED** [review]
Looks clean now. Just a couple comments.

File: ARCharts/ARBarChart.swift (hunk 4 above)
Comment (Boris-Em) [reply ARCharts/ARBarChart.swift]: Duh...

**Pull request merged**